pkg/pipeline/providers/gitlab/provider.go

```diff
@@ -104,3 +76,25 @@ func (g *GlProvider) GetProviderConfig(projectID string) (interface{}, error) {
 	storedGitlabPipelineConfig.Kind = v3.SourceCodeProviderConfigGroupVersionKind.Kind
 	return storedGitlabPipelineConfig, nil
 }
+
+func formGitlabRedirectURLFromMap(config map[string]interface{}) string {
+	hostname := convert.ToString(config[client.GitlabPipelineConfigFieldHostname])
+	clientID := convert.ToString(config[client.GitlabPipelineConfigFieldClientID])
+	tls := convert.ToBool(config[client.GitlabPipelineConfigFieldTLS])
+	return gitlabRedirectURL(hostname, clientID, tls)
+}
+
+func gitlabRedirectURL(hostname, clientID string, tls bool) string {
+	redirect := ""
+	if hostname != "" {
+		scheme := "http://"
+		if tls {
+			scheme = "https://"
+		}
+		redirect = scheme + hostname
+	} else {
+		redirect = gitlabDefaultHostName
+	}
+	redirect = fmt.Sprintf("%s/oauth/authorize?client_id=%s&response_type=code", redirect, clientID)
```

Comment:
Nit: we can just `return fmt...` as redirect variable is not being used after that